extensions/rvadapter/src/main/java/com/google/android/agera/rvadapter/RepositoryPresenterCompilerStates.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.agera.rvadapter;
+
+import com.google.android.agera.Binder;
+import com.google.android.agera.Function;
+import com.google.android.agera.Receiver;
+import com.google.android.agera.Repository;
+import com.google.android.agera.Result;
+
+import android.support.annotation.LayoutRes;
+import android.support.annotation.NonNull;
+import android.support.v7.widget.RecyclerView;
+import android.view.View;
+
+import java.util.List;
+
+/**
+ * Container of the compiler state interfaces supporting the creation of a
+ * {@link RepositoryPresenter}.
+ */
+public interface RepositoryPresenterCompilerStates {
+
+  /**
+   * Compiler state to specify what layout to use.
+   */
+  interface RPLayout<TVal, TRet> {
+
+    /**
+     * Specifies the {@code layoutId} for the layout to inflate for each item in the associated
+     * {@link Repository}.
+     */
+    @NonNull
+    TRet layout(@LayoutRes int layoutId);
```

Comment:
Right now it only allows for either static or dynamic, not both at the same time. The idea was that if you'd need a fallback it's probably better to return a default from `layoutForItem` rather than return 0 and then fallback to default?